pageserver/src/layered_repository.rs

```diff
@@ -1638,34 +1508,273 @@ impl LayeredTimeline {
         Ok(())
     }
 
-    fn evict_layer(
+    pub fn compact(&self) -> Result<()> {
```

Comment:
I wonder why it is named "compact" while it is actually performs reshuffling and generation of image layers?
So size of the storage after this operation is not reduced, but only increased.